src/main/generators/OpenApiGenerator.ts

```diff
@@ -77,15 +82,45 @@ export class OpenApiGenerator {
                         throw new UnsupportedFeatureError('Unsupported request body content');
                     }
                 }
-                yield {
-                    moduleName: inferredModuleName,
-                    method: ep.method,
-                    path: normalizePath(ep.path),
-                    description: ep.opSpec.description ?? ep.opSpec.summary ?? '',
-                    externalDocs: ep.opSpec.externalDocs?.url ?? '',
-                    params: paramSpecs,
-                    requestBodyType,
-                };
+
+                const existingModule = existingModules.find(mod => mod.operationId === ep.opSpec.operationId);
+
+                if (existingModule) {
+                    const mergedParams = paramSpecs.map(newParam => {
+                        const existingParam = existingModule.params.find(p => p.paramName === newParam.paramName);
+                        if (!existingParam) {
+                            return newParam;
+                        }
+                        return {
+                            ...existingParam,
+                            description: newParam.description || existingParam.description,
+                            schema: existingParam.schema,
+                            required: existingParam.required,
```

Comment:
I think these two are already on `existingParam` - or are they supposed to be like `newParam.schema || existingParam.schema?